### DISCUSSION TITLE
[Question] Why is the click event handler blocking propagation?

### DISCUSSION THREAD

**Discussion by arthuro555** [4ian/GDevelop#3194] [Archive] [answered]
My question is pretty straightforward: why is the click event blocking propagation? It is really annoying because I wanted a user to select a file from an HTML game, so I created a file input element via the dom and triggered the click event with `domNode.click()`. It didn't open, but the exact same code worked when pasted in the inspector of a non GDevelop game website. After searching it seems the problem is that the click method triggers click event but propagation to the Dom node is blocked after GDevelop's one blocked propagation. Why is it designed that way? 
The concerned code:
https://github.com/4ian/GDevelop/blob/661d32917002dd4f2852c67b22f62c294e99bd8d/GDJS/Runtime/pixi-renderers/runtimegame-pixi-renderer.js#L363

**Comment by Bouh** [comment, marked as answer]
Maybe for avoid to select/hightlight the canvas in the webview and other things.

**Comment by 4ian** [comment]
> Maybe for avoid to select/hightlight the canvas in the webview and other things.

Yes exactly.

Though if you think of a better solution or if it's useless, feel free to try to remove it and see how it goes :)

**Comment by arthuro555** [comment]
Maybe removing not removing, but at least some way to disable it, or to put event handlers before it. Maybe by making the function a static function of the runtimegame renderer, so that I can use removeEventListener and then addEventListener again.

**Comment by 4ian** [comment]
I would recommend that you for now hack/remove it and see if you can have something working in your case. I'm not sure what is the best solution (temporarily remove the prevent default on demand, always remove it, make it configurable, etc...) and well know more with a use case :)

**Comment by arthuro555** [comment]
The problem is there is no way I can remove it without having the original function handling the event.

________________________________
